session.go

```diff
@@ -13,6 +13,10 @@ type SessionMaker interface {
 
 // NewSession returns a session backed NodeGetter if the given NodeGetter
 // implements SessionMaker.
+//
+// Deprecated: Use exchange.NewSession to wrap your context. This function only
+// works if the NodeGetter happens to implement the session interface. If it has
+// been wrapped in some way, it won't work.
```

Comment:
Can you specify how you can then get the session out of the context afterwards too?

Might be worth changing the signature and returning an error when the NodeGetter is not a SessionMaker. People might be using this wrong and the deprecation notice will not really fix their code. Thinking that someone is using this wrong without deep-auditing their code will be difficult for us if they ever ask, and this might be a major perf sink for them even in the simplest application.

Reply:
Would a link to the NewSession docs (and some examples there) suffice? I'd like to avoid duplicating information.

Reply:
Yes for docs.